plugins/ida/python/bap_utils.py

```diff
@@ -0,0 +1,145 @@
+"""Commonly used utilities for BAP IDAPython plugins to use."""
+
+import idaapi
+
+
+def sexp2list(s):
+    """Convert S-Expression to List."""
+    sexp = [[]]
+    word = ''
+    in_str = False
+    for c in s:
+        if c == '(' and not in_str:
+            sexp.append([])
+        elif c == ')' and not in_str:
+            if word:
+                sexp[-1].append(word)
+                word = ''
+            temp = sexp.pop()
+            sexp[-1].append(temp)
+        elif c in (' ', '\n', '\t') and not in_str:
+            if word:
+                sexp[-1].append(word)
+            word = ''
+        elif c == '\"':
+            in_str = not in_str
+        else:
+            word += c
+    return sexp[0]
+
+
+def list2sexp(l):
+    """Convert List to S-Expression."""
+    if isinstance(l, str):
+        for special_char in (' ', '\n', '\t', '(', ')', '\"'):
+            if special_char in l:
+                return '\"' + l + '\"'
+        return l
+    return '(' + ' '.join(list2sexp(e) for e in l) + ')'
+
+
+def get_bap_comment(comm):
+    """
+    Get '(BAP )' style comment from given string.
+
+    Returns tuple (BAP_dict, start_loc, end_loc)
+        BAP_dict: The '(BAP )' style comment
+        start_loc: comm[:start_loc] was before the BAP comment
+        end_loc: comm[end_loc:] was after the BAP comment
+    """
+    if '(BAP ' in comm:
```

Comment:
Am I right, that this algorithm assumes that value inside the sexp will be a well-formed sexp? E.g., something like this `(BAP (message "hello, world))"))` is not allowed?